### ISSUE TITLE
Fix data leakage 

### ISSUE THREAD

**Issue by angelmtenor** [angelmtenor/data-science-keras#1]
- Do not use test data when scaling
- Check highly correlated features in exploratory analysis
- Prepare Data in each CV fold separately

- Hide test validation set until the end

**Comment by angelmtenor** [comment]
Old Issue - Fixed in all new repos by using Scikit-learn Pipelines
